tensorflow/compiler/xla/service/pattern_matcher_test.cc

```diff
@@ -1037,6 +1040,62 @@ TEST_F(PatternMatcherTest, OneUseAndOneUser) {
             "in p0 = f32[] parameter(0)");
 }
 
+TEST_F(PatternMatcherTest, MatchSingleUserOperands) {
+  auto param =
+      HloInstruction::CreateParameter(0, ShapeUtil::MakeShape(F32, {}), "p");
+
+  auto reshape =
+      SetName("reshape", HloInstruction::CreateReshape(
+                             ShapeUtil::MakeShape(F32, {1}), param.get()));
+  EXPECT_TRUE(MatchSingleUserOperands(reshape.get(), m::Reshape(m::Op())));
+
+  auto bitcast =
+      SetName("bitcast", HloInstruction::CreateBitcast(
+                             ShapeUtil::MakeShape(F32, {1}), param.get()));
+  EXPECT_TRUE(MatchSingleUserOperands(param.get(), m::Op()));
+  EXPECT_TRUE(MatchSingleUserOperands(bitcast.get(), m::Bitcast()));
+  EXPECT_FALSE(MatchSingleUserOperands(bitcast.get(), m::Bitcast(m::Op())));
+  EXPECT_EQ(Explanation(bitcast.get(), m::Bitcast(m::Op()), true),
```

Comment:
Could you break those apart into subtests and contrast with Match() API for clarity?